apps/src/gamelab/GameLabSprite.js

```diff
@@ -552,21 +552,32 @@ var isTouching = function (p5Inst, target) {
           }
           else //non tunnel overlap
           {
+            /* First, check for adjacency using overlap method.
+             *
+             * We do this to ensure isTouching returns true for cases when
+             * bounce, collide would return true and also when overlap would
+             * return true.
+             *
+             * NOTE: this.touching will remain all false in the adjacent case,
+             * but the function will return true.
+             */
 
             //if the other is a circle I calculate the displacement from here
             //and reverse it
-            if(this.collider instanceof CircleCollider)
-            {
-              displacement = other.collider.collide(this.collider).mult(-1);
+            if (this.collider instanceof CircleCollider) {
+              if (other.collider.overlap(this.collider)) {
+                result = true;
+                displacement = other.collider.collide(this.collider).mult(-1);
+              }
             }
-            else
+            else if (this.collider.overlap(other.collider)) {
+              result = true;
               displacement = this.collider.collide(other.collider);
-
+            }
```

Comment:
I was going to mention how redundant this logic looks, but then I did some digging and learned why it's written this way...

* `this.collider` and `other.collider` can each be either a `CircleCollider` or an `AABB`.
* `CircleCollider.overlap()` only works if passed a `CircleCollider` argument.
* `AABB.overlap()` can take either `CircleCollider` or `AABB` as an argument.

Therefore, it's only safe to call `a.overlap(b)` if `(a instanceof AABB) || (b instanceof CircleCollider)`.

:sob: